server/neptune/workflows/internal/deploy/revision/queue/deployer_test.go

```diff
@@ -94,233 +99,321 @@ func testDeployerWorkflow(ctx workflow.Context, r deployerRequest) (*deployment.
 
-	env.OnActivity(da.StoreLatestDeployment, mock.Anything, storeDeploymentRequest).Return(nil)
+			repo := github.Repo{
+				Owner: "owner",
+				Name:  "test",
+			}
 
-	env.ExecuteWorkflow(testDeployerWorkflow, deployerRequest{
-		Info: deploymentInfo,
-	})
+			root := model.Root{
+				Name: "root_1",
+			}
 
-	env.AssertExpectations(t)
+			deploymentInfo := terraform.DeploymentInfo{
+				ID:         uuid.UUID{},
+				Revision:   "3455",
+				CheckRunID: 1234,
+				Root:       root,
+				Repo:       repo,
+			}
 
-	var resp *deployment.Info
-	err := env.GetWorkflowResult(&resp)
-	assert.NoError(t, err)
+			if c.setPRRevision {
+				env.RegisterWorkflow(testPRRevWorkflow)
+				env.OnWorkflow(testPRRevWorkflow, mock.Anything, prrevision.Request{
+					Repo:     repo,
+					Root:     root,
+					Revision: deploymentInfo.Revision,
+				}).Return(nil)
+			} else {
+				env.OnGetVersion(version.SetPRRevision, workflow.DefaultVersion, 1).Return(workflow.DefaultVersion)
+			}
 
-	assert.Equal(t, latestDeployedRevision, resp)
-}
+			latestDeployedRevision := &deployment.Info{
+				ID:       deploymentInfo.ID.String(),
+				Version:  1.0,
+				Revision: "3455",
+				Root: deployment.Root{
+					Name: deploymentInfo.Root.Name,
+				},
+				Repo: deployment.Repo{
+					Owner: deploymentInfo.Repo.Owner,
+					Name:  deploymentInfo.Repo.Name,
+				},
+			}
 
-func TestDeployer_CompareCommit_DeployAhead(t *testing.T) {
-	ts := testsuite.WorkflowTestSuite{}
-	env := ts.NewTestWorkflowEnvironment()
+			storeDeploymentRequest := activities.StoreLatestDeploymentRequest{
+				DeploymentInfo: &deployment.Info{
+					Version:  deployment.InfoSchemaVersion,
+					ID:       deploymentInfo.ID.String(),
+					Revision: deploymentInfo.Revision,
+					Root: deployment.Root{
+						Name: deploymentInfo.Root.Name,
+					},
+					Repo: deployment.Repo{
+						Owner: deploymentInfo.Repo.Owner,
+						Name:  deploymentInfo.Repo.Name,
+					},
+				},
+			}
 
-	da := &testDeployActivity{}
-	env.RegisterActivity(da)
+			env.OnActivity(da.StoreLatestDeployment, mock.Anything, storeDeploymentRequest).Return(nil)
 
-	repo := github.Repo{
-		Owner: "owner",
-		Name:  "test",
-	}
+			env.ExecuteWorkflow(testDeployerWorkflow, deployerRequest{
+				Info: deploymentInfo,
+			})
 
-	root := model.Root{
-		Name: "root_1",
-	}
+			env.AssertExpectations(t)
 
-	deploymentInfo := terraform.DeploymentInfo{
-		ID:         uuid.UUID{},
-		Revision:   "3455",
-		CheckRunID: 1234,
-		Root:       root,
-		Repo:       repo,
+			var resp *deployment.Info
+			err := env.GetWorkflowResult(&resp)
+			assert.NoError(t, err)
+
+			assert.Equal(t, latestDeployedRevision, resp)
+		})
 	}
+}
 
-	latestDeployedRevision := &deployment.Info{
-		ID:       deploymentInfo.ID.String(),
-		Version:  1.0,
-		Revision: "3255",
-		Root: deployment.Root{
-			Name: deploymentInfo.Root.Name,
+func TestDeployer_CompareCommit_DeployAhead(t *testing.T) {
+	cases := []struct {
+		description   string
+		setPRRevision bool
```

Comment:
doing this makes it hard to remove. Can we just create separate tests instead of this boolean logic? Then it's just a big deletion block when we want to remove it.

Reply:
You don't need to duplicate all the tests, you just need to have one test that asserts that the child workflow is called.

Reply:
That's fair! I was originally duplicating all the tests that would end up starting the rebase workflow which was a lot of duplicated code, but I guess I can only have one that asserts the child workflow is called.